src/core_modules/capture-core/components/WidgetEventSchedule/WidgetEventSchedule.container.js

```diff
@@ -41,28 +35,35 @@ export const WidgetEventSchedule = ({
     const [scheduleDate, setScheduleDate] = useState(suggestedScheduleDate);
     const { events } = useEventsInOrgUnit(orgUnitId, scheduleDate);
 
-    const [mutate] = useDataMutation(scheduleEventMutation, {
-        onComplete: () => {
-            history.push(`/enrollment?${urlArguments({ orgUnitId, programId, stageId, enrollmentId })}`);
-        },
-    });
     const eventCountInOrgUnit = events
         .filter(event => moment(event.dueDate).format('YYYY-MM-DD') === scheduleDate).length;
 
-    const onHandleSchedule = async () => {
-        if (!scheduleDate) { return; }
-        await mutate({ events: [{
-            dueDate: scheduleDate,
-            dataValues: [],
-            trackedEntityInstance: teiId,
-            orgUnit: orgUnitId,
-            enrollment: enrollmentId,
-            program: programId,
-            programStage: stageId,
-            status: 'SCHEDULE',
-            notes: [],
-        }] });
-    };
+    useEffect(() => {
+        dispatch(startEditingScheduleEvent());
+    }, []); // eslint-disable-line
+
+    useEffect(() => {
+        if (!scheduleDate && suggestedScheduleDate) { setScheduleDate(suggestedScheduleDate); }
+    }, [suggestedScheduleDate]); // eslint-disable-line
```

Comment:
Same doubt here, why don't you pass all the dependencies to the array? Thanks!

Reply:
hey, because I only what to update the schedule date once when on mount. Thanks